### ISSUE TITLE
Email addresses should not probably be translatable

### ISSUE THREAD

**Issue by ghost** [HabitRPG/habitica#8385]
Following the discussion in the i18 guild...

> ***Alys***  Also for Czech translators ... : In the translations for the Community Guidelines (communityguidelines.json), there's some problems with email addresses and URLs. In at least two places, the leslie@habitica.com email address has been written incorrectly in both the mailto: portion and visible portion of email links (for example, <a href=\"mailto:leslie@\">leslie@habitica</a>)...

Although I haven't managed to find mistakes in the mailto part of the strings, for consistency and security reasons may we have email addresses as tokens so that translators could not change them?

E.g.: instead of
`If you have an issue or concern about a particular Mod, please send an email to Lemoness (<a href="mailto:leslie@habitica.com">leslie@habitica.com</a>).`
introduce something like
`If you have an issue or concern about a particular Mod, please send an email to Lemoness (<%= email %>).`

FYI @Alys @paglias 

----

_edit by Alys 2017-01-16:_

We'll use environment variables (i.e., new items in config.json) to specify the email addresses and then use tokens in the locales strings to specify where the addresses should be placed. See the comments below for more details.

**Comment by Alys** [comment]
That's a really good idea!

The <%= email %> token could be used to insert a value from an environment
variable so that we could change it easily when needed (e.g., <%=
communityManagerEmail %> and change the value when/if Lemoness goes on
holiday).

If there's no disagreements for this idea, I can make a list of all current
addresses in the strings and then we can plan suitable variable names for
them.

Also I'll track down the incorrect mailto links that (I think) I found.

**Comment by khipkin** [comment]
Suggestion: [translations] and not sure if medium- or entry-level-coding?

**Comment by Alys** [comment]
Below are all the locales strings that contain email addresses, separated into sections based on what the email addresses are used for, with my suggestions for the environment variable names that we use for each address.

@paglias @SabreCat  Do you have any objections to using ENV variables for this? Or better ideas?

----
suggested ENV variable: `PRESS_ENQUIRY_EMAIL`
current value: leslie@habitica.com

```
front.json:    "presskitText": "Thanks for your interest in Habitica! The following images can be used for articles or videos about Habitica. For more information, please contact Siena Leslie at leslie@habitica.com.",
```

----

suggested ENV variable: `TECH_ASSISTANCE_EMAIL`
current value: admin@habitica.com

```
faq.json:  "webFaqAnswer11": "To report a bug, go to [Help > Report a Bug](https://habitica.com/#/options/groups/guilds/a29da26b-37de-4a71-b0c6-48e72a900dac) and read the points above the chat box. If you're unable to log in to Habitica, send your login details (not your password!) to [admin@habitica.com](mailto:admin@habitica.com). Don't worry, we'll get you fixed up soon!\n<br><br>\n Feature requests are collected on Trello. Go to [Help > Request a Feature](https://trello.com/c/odmhIqyW/440-read-first-table-of-contents) and follow the instructions. Ta-da!",

front.json:    "cannotFulfillReq":"Your request cannot be fulfilled.  Email admin@habitica.com if this error persists.",

maintenance.json:  "maintenanceInfoAccountChangesText": "At first, there won't be any notable changes aside from performance improvements for features such as Challenges. If you notice any changes that shouldn't be there, email us at <a href='mailto:admin@habitica.com'>admin@habitica.com</a> and we will investigate them for you!",

settings.json:    "APITokenWarning": "If you need a new API Token (e.g., if you accidentally shared it), email <a href='mailto:admin@habitica.com'>admin@habitica.com</a> with your User ID and current Token. Once it is reset you will need to re-authorize everything by logging out of the website and mobile app and by providing the new Token to any other Habitica tools that you use.",
```

----

suggested ENV variable: `COMMUNITY_MANAGER_EMAIL`
current value: leslie@habitica.com

```
character.json:    "communityGuidelinesWarning": "Please keep in mind that your Display Name, profile photo, and blurb must comply with the <a href='https://habitica.com/static/community-guidelines' target='_blank'>Community Guidelines</a> (e.g. no profanity, no adult topics, no insults, etc). If you have any questions about whether or not something is appropriate, feel free to email <a href='mailto:leslie@habitica.com' target='blank'>leslie@habitica.com</a>!",

communityGuidelines.json:    "commGuidePara012": "If you have an issue or concern about a particular Mod, please send an email to Lemoness (<a href=\"mailto:leslie@habitica.com\">leslie@habitica.com</a>).",

communityGuidelines.json:    "commGuideList02H": "<strong>Divisive/contentious conversations should be reported to mods.</strong> If you feel that a conversation is getting heated, overly emotional, or hurtful, cease to engage. Instead, email <a href=\"mailto:leslie@habitica.com\">leslie@habitica.com</a> to let us know about it. It's our job to keep you safe.",

communityGuidelines.json:    "commGuidePara033": "<strong>Public Guilds may NOT contain 18+ content. If they plan to regularly discuss sensitive content, they should say so in the Guild title.</strong> This is to keep Habitica safe and comfortable for everyone. If the guild in question has different kinds of sensitive issues, it is respectful to your fellow Habiticans to place your comment behind a warning (ex. \"Warning: references self-harm\"). Additionally, the sensitive material should be topical -- bringing up self-harm in a guild focused on fighting depression may make sense, but may be less appropriate in a music guild. If you see someone who is repeatedly violating this guideline, even after several requests, please email <a href=\"mailto:leslie@habitica.com\">leslie@habitica.com</a> with screenshots.",

communityGuidelines.json:    "commGuideList06A": "Ignoring or Disrespecting a Mod. This includes publicly complaining about moderators or other users/publicly glorifying or defending banned users. If you are concerned about one of the rules or Mods, please contact Lemoness via email (<a href=\"mailto:leslie@habitica.com\">leslie@habitica.com</a>).",

communityGuidelines.json:    "commGuidePara067": "So there you have it, brave Habitican -- the Community Guidelines! Wipe that sweat off of your brow and give yourself some XP for reading it all. If you have any questions or concerns about these Community Guidelines, please email Lemoness (<a href=\"mailto:leslie@habitica.com\">leslie@habitica.com</a>) and she will be happy to help clarify things.",

contrib.json:    "blurbInbox": "This is where your private messages are stored! You can send someone a message by clicking on the envelope icon next to their name in Tavern, Party, or Guild Chat. If you've received an inappropriate PM, you should email a screenshot of it to Lemoness (<a href=\"mailto:leslie@habitica.com\">leslie@habitica.com</a>)",

front.json:    "accountSuspended": "Account has been suspended, please contact leslie@habitica.com with your User ID \"<%= userId %>\" for assistance.",
```

----

**Comment by paglias** [comment]
@Alys that's ok for me!

**Comment by Alys** [comment]
I've edited my previous comment to change the suggested names for the environment variables from camelCase to all caps, for consistency with all other ENV vars (e.g., PRESS_ENQUIRY_EMAIL instead of pressEnquiryEmail)

**Comment by gsacavdm** [comment]
Hi @Alys, I can certainly help with this. I can easily update the strings in the json files as follows:

    communityGuidelines.json:    "commGuidePara012": "If you have an issue or concern about a particular Mod, please send an email to Lemoness <%= COMMUNITY_MANAGER_EMAIL =>)."

And the jade files as follows:

    p!=env.t('commGuidePara012', { "COMMUNITY_MANAGER_EMAIL": env.COMMUNITY_MANAGER_EMAIL })

The only thing I'm unclear about is the where to define these environment variables.

PS: Should I be asking about this here or in the Guild chat? N00b here :)

**Comment by Alys** [comment]
@gsacavdm It's best to ask questions about a specific issue in the issue, as you've done here. That keeps all the information together.

In the your local install, the environment variables are in the config.json file. When you're ready to make a pull request, you'd copy them to the config.json.example file and commit that file along with your code changes. You wouldn't commit the config.json file (because it can be changed in customised ways for each local install). On the habitica.com production webserver, the environment variables are stored in the server's config and an admin would set them up when they were ready to deploy your PR.

Thank you!

**Comment by gsacavdm** [comment]
Thanks for the tips @Alys, I've started the work on this and have pushed it to my fork. I've added comments to the commit. If you or others can take a quick look to make sure I'm going the right direction, that would be great.

Here's the link to my commit:
https://github.com/gsacavdm/habitica/commit/6e590f8edd8ed12da5fd93eae34f06fecc32be68

Reiterate my disclaimer: Still a n00b, so I apologize if I'm do any faux-pas, happy to get called out on them and improve.
